### PR TITLE
otelcolconvert: support converting memory_limiter processor

### LINKED ISSUE #6453: Support converting `memorylimiterprocessor`

### PR DESCRIPTION
**Pull request by tpaschalis**
Fixes #6453

### CODE REVIEW

File: converter/internal/otelcolconvert/testdata/memorylimiter.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # defaults below (balancer_name and queue_size).
+    endpoint: database:4317
+    balancer_name: pick_first
+    sending_queue:
+      queue_size: 5000
+
+processors:
+  memory_limiter:
+    check_interval: 1s
+    limit_mib: 147
```
Comment (rfratto) [inline converter/internal/otelcolconvert/testdata/memorylimiter.yaml:19]: @tpaschalis Are these out of sync with upstream? Can you create an issue to resync our defaults similar to the other ones I created?

Comment (rfratto) [reply converter/internal/otelcolconvert/testdata/memorylimiter.yaml:19]: NVM I see that they are not the defaults of upstream

Comment (tpaschalis) [reply converter/internal/otelcolconvert/testdata/memorylimiter.yaml:19]: No, I just added them to make sure that everything was going through properly. I'll remove them.

Comment (rfratto) [reply converter/internal/otelcolconvert/testdata/memorylimiter.yaml:19]: I think it's fine to keep them in as a good test, I was just thrown off :)

**Pull request merged**